Sources/NIOEmbedded/AsyncTestingEventLoop.swift

```diff
@@ -95,6 +95,10 @@ public final class NIOAsyncTestingEventLoop: EventLoop, @unchecked Sendable {
     /// The queue on which we run all our operations.
     private let queue = DispatchQueue(label: "io.swiftnio.AsyncEmbeddedEventLoop")
 
+    private enum State: Int, AtomicValue { case open, closing, closed }
+    private let _state = ManagedAtomic(State.open)
+    private var state: State { self._state.load(ordering: .relaxed) }
```

Comment:
The atomic ordering here seems likely to be logically wrong. We should probably force ourselves to express it at all points.

Reply:
Sounds good. I saw a bunch of `.relaxed` in this implementation already but I think they're probably OK. I realise that this isn't appropriate here now. Updated.